### PR TITLE
[jest][express][1/x] Converted old unit tests into Jest testing framework

### PR DESCRIPTION
**Pull request by amotor95**
### Summary
- Converted the following tests to use the Jest unit testing framework:
   - UserTests
   - ComparabilityScoresTests
   - ComparablePartsTests
   - EbayScraperTests
   - EvaluationTests
   - BuildRecommendationTests
   - GenerateBuildsTest
- Added logging parameter to getRecentlySoldListings to turn logging on/off

### Tests
|Running npm test|
|---|
|image|

### CODE REVIEW

**Comment by auroraw9825** [comment]
i would recommend adding negative or edge case tests, such as empty list, mismatched keys, invalid spec types, etc

File: server/tests/builds/build_recommender.test.js [hunk 1]
```diff
@@ -41,24 +35,16 @@ const calcSlidingQualityRatingTester = (test_allocations, test_ratings, parts, c
...
             const prev_part_rating = sortedParts[part_index-1][spec_type]
             const prev_part_rating_index = ratings.indexOf(prev_part_rating)
-            if (prev_part_rating_index === -1) {
-                throw new TestError(`test_${component_type}_builds::${spec_type}_allocation - Failed`)
-            }
-            if (prev_part_rating_index > part_rating_index && sortedParts[part_rating_index]['model'] !== 'form_factor') {
-                throw new TestError(`test_${component_type}_builds::${spec_type}_allocation - Failed`)
-            }
+            expect(prev_part_rating_index).toBeGreaterThan(-1)
+            expect(prev_part_rating_index > part_rating_index && sortedParts[part_rating_index]['model'] !== 'form_factor').toBe(false)
```
Comment (auroraw9825) [inline server/tests/builds/build_recommender.test.js:42]: consider 
```
if (sortedParts[part_rating_index]['model'] !== 'form_factor') {
  expect(prev_part_rating_index).toBeLessThanOrEqual(part_rating_index)
}
```

File: server/tests/parts/evaluation.test.js [hunk 2]
```diff
@@ -117,58 +116,27 @@ const outlier_test_listings_key = [
...
-        removePriceOutliers_test()
-    } catch (error) {
-        console.error(error)
-    }
-    process.exit(0)
-}
-
-running_tests()
\ No newline at end of file
+    expect(result.sort()).toEqual(outlier_test_listings_key.sort())
```
Comment (auroraw9825) [inline server/tests/parts/evaluation.test.js:141]: You’re calling `sort(...)` inside tests.
if you are testing function `removePriceOutliers`, then in the test, you should call `xxx.removePriceOutliers()` instead of it's inner logic

**Review by auroraw9825 — COMMENTED** [review]
recommend using `describe` to organize the test file into sections per component:

```
describe('CPU rating system', () => {
  test('ranks by cores correctly', () => {})
  test('ranks by base_clock correctly', () => {})
})
```

File: server/tests/parts/evaluation.test.js (hunk 2 above)
Comment (amotor95) [reply server/tests/parts/evaluation.test.js:141]: Explained earlier but I'll leave a note,
the sort is unrelated to removePriceOutliers and simply to make sure the output array from removePriceOutliers and my key array are in the same order for comparison.
